### ISSUE TITLE
Dynamic edge convolution (NNConv) vs graph attention network (GATConv)

### ISSUE THREAD

**Issue by adhikarirsr** [pyg-team/pytorch_geometric#1942]
Isn't GATConv extension of NNConv? I mean GATConv has attention heads, which are multiple filters that you learn using neural nets instead of a filter for every edge you learn using neural nets for NNConv.

**Comment by rusty1s** [comment]
Tricky question. In general, I would argue that `NNConv` is not an attention-based GNN. `GATConv` performs weighted aggregation (it learns a scalar for each incoming message), while `NNConv` learns a complete individual weight matrix for each incoming message. The difference is that `GATConv` transforms neighboring features the same (they are just scaled differently), while `NNConv` does not.

**Comment by ldv1** [comment]
NNConv and GATConv are both anisotropic message passing schemes. NNConv can use the GATConv additive attention mechanism (take the edge attribute e_{ij} = [ x_i, x_j ]), but it will not use at the end the softmax "glue" that you find in GATConv. From this perspective (I guess this is what rusty1s says), it is not considered an attention-based GNN.

**Comment by adhikarirsr** [comment]
I agree. Also in `GATConv`, we can have multiple attention heads.
